spokestack/nlu/result.py

```diff
@@ -0,0 +1,56 @@
+"""
+This module contains the Result class for the NLU.
+"""
+from typing import Any, Dict
+
+from spokestack.typing import Intent, Slot, Utterance
+
+
+class Result:
+    """ Convenience wrapper for NLU Results"""
+
+    def __init__(self):
```

Comment:
I think it would be nice to construct the result and pass all of its fields at once

that way, you can make the class immutable (no writable properties) and you can remove the `Union` on all of the type specs, so that the client doesn't have to worry about `None` values